src/db/commit.rs

```diff
@@ -7,6 +7,40 @@ use str_macro::str;
 
 pub const DEFAULT_HEAD_NAME: &str = "main";
 
+#[derive(Clone, Copy)]
+enum Ref<'a> {
+    None,
```

Comment:
Might be slightly more rusty be to have the enum have Strong and Weak and the field that holds it be an Option<Ref>.

Reply:
Thanks. That did make things simpler since we can reuse all the rust building blocks